File: vue-version/scan-files.py
#!/usr/bin/env python3
import os
import json
from pathlib import Path
from datetime import datetime

BIJI_DIR = Path(__file__).parent / 'biji'
OUTPUT_FILE = Path(__file__).parent / 'file-list.json'

def scan_directory(directory, parent_path=''):
    items = []
    
    try:
        entries = sorted(directory.iterdir(), key=lambda x: (not x.is_dir(), x.name.lower()))
    except PermissionError:
        return items
    
    for entry in entries:
        if entry.name.startswith('.'):
            continue
            
        relative_path = f"{parent_path}/{entry.name}" if parent_path else entry.name
        
        if entry.is_dir():
            children = scan_directory(entry, relative_path)
            if children:
                items.append({
                    'name': entry.name,
                    'path': relative_path,
                    'isFolder': True,
                    'children': children
                })
        elif entry.suffix.lower() in ['.md', '.markdown']:
            stat = entry.stat()
            items.append({
                'name': entry.name,
                'path': relative_path,
                'isFolder': False,
                'size': stat.st_size,
                'updatedAt': stat.st_mtime
            })
    
    return items

def main():
    print(f"Scanning directory: {BIJI_DIR}")
    
    if not BIJI_DIR.exists():
        print(f"❌ Directory not found: {BIJI_DIR}")
        print(f"   Please create the 'biji' folder and add your markdown files")
        return
    
    tree = scan_directory(BIJI_DIR)
    
    output = {
        'version': '1.0',
        'timestamp': datetime.now().timestamp(),
        'tree': tree
    }
    
    with open(OUTPUT_FILE, 'w', encoding='utf-8') as f:
        json.dump(output, f, ensure_ascii=False, indent=2)
    
    print(f"✅ File list generated: {OUTPUT_FILE}")
    print(f"📁 Found {len(tree)} top-level items")
    
    file_count = sum(1 for item in tree if not item.get('isFolder'))
    folder_count = sum(1 for item in tree if item.get('isFolder'))
    print(f"   📄 {file_count} files, 📁 {folder_count} folders")

if __name__ == '__main__':
    main()

File: vue-version/scan-files.js
import fs from 'fs';
import path from 'path';

// 获取当前脚本所在目录
const getDirname = () => {
    const url = new URL(import.meta.url);
    let dirname = path.dirname(url.pathname);
    
    // 处理 Windows 路径
    if (process.platform === 'win32') {
        // 移除前导斜杠
        if (dirname.startsWith('/')) {
            dirname = dirname.slice(1);
        }
        // 转换为 Windows 路径分隔符
        dirname = dirname.replace(/\//g, '\\');
    }
    
    return dirname;
};

const CURRENT_DIR = getDirname();
const BIJI_DIR = path.join(CURRENT_DIR, 'biji');
const OUTPUT_FILE = path.join(CURRENT_DIR, 'file-list.json');

function scanDirectory(directory, parentPath = '') {
    const items = [];
    
    try {
        const entries = fs.readdirSync(directory, { withFileTypes: true });
        
        // 排序：文件夹在前，文件在后，按名称排序
        entries.sort((a, b) => {
            const aIsDir = a.isDirectory();
            const bIsDir = b.isDirectory();
            
            if (aIsDir && !bIsDir) return -1;
            if (!aIsDir && bIsDir) return 1;
            
            return a.name.toLowerCase().localeCompare(b.name.toLowerCase());
        });
        
        for (const entry of entries) {
            if (entry.name.startsWith('.')) {
                continue;
            }
            
            const relativePath = parentPath ? `${parentPath}/${entry.name}` : entry.name;
            const fullPath = path.join(directory, entry.name);
            
            if (entry.isDirectory()) {
                const children = scanDirectory(fullPath, relativePath);
                if (children.length > 0) {
                    items.push({
                        'name': entry.name,
                        'path': relativePath,
                        'isFolder': true,
                        'children': children
                    });
                }
            } else if (['.md', '.markdown'].includes(path.extname(entry.name).toLowerCase())) {
                const stat = fs.statSync(fullPath);
                items.push({
                    'name': entry.name,
                    'path': relativePath,
                    'isFolder': false,
                    'size': stat.size,
                    'updatedAt': stat.mtime.getTime()
                });
            }
        }
    } catch (err) {
        console.error(`Error scanning directory: ${err.message}`);
    }
    
    return items;
}

export function main() {
    console.log(`Scanning directory: ${BIJI_DIR}`);
    
    if (!fs.existsSync(BIJI_DIR)) {
        console.log(`❌ Directory not found: ${BIJI_DIR}`);
        console.log(`   Please create the 'biji' folder and add your markdown files`);
        return;
    }
    
    const tree = scanDirectory(BIJI_DIR);
    
    const output = {
        'version': '1.0',
        'timestamp': Date.now(),
        'tree': tree
    };
    
    fs.writeFileSync(OUTPUT_FILE, JSON.stringify(output, null, 2), 'utf-8');
    
    console.log(`✅ File list generated: ${OUTPUT_FILE}`);
    console.log(`📁 Found ${tree.length} top-level items`);
    
    const fileCount = tree.filter(item => !item.isFolder).length;
    const folderCount = tree.filter(item => item.isFolder).length;
    console.log(`   📄 ${fileCount} files, 📁 ${folderCount} folders`);
}

// 直接运行时执行
if (process.argv[1] && process.argv[1].endsWith('scan-files.js')) {
    main();
}
